Source: Raphael Kubo da Costa <rakuco@FreeBSD.org>
Upstream: backported from 4.9 branch
Reason: fix automagic dependency on media-libs/libcanberra

diff -urN a/cmake/modules/FindCanberra.cmake b/cmake/modules/FindCanberra.cmake
--- a/cmake/modules/FindCanberra.cmake	1970-01-01 01:00:00.000000000 +0100
+++ b/cmake/modules/FindCanberra.cmake	2012-08-26 11:50:44.302063991 +0200
@@ -0,0 +1,29 @@
+# - Find libcanberra's libraries and headers.
+# This module defines the following variables:
+#
+#  CANBERRA_FOUND        - true if libcanberra was found
+#  CANBERRA_LIBRARIES    - libcanberra libraries to link against
+#  CANBERRA_INCLUDE_DIRS - include path for libcanberra
+#
+# Copyright (c) 2012 Raphael Kubo da Costa <rakuco@FreeBSD.org>
+#
+# Redistribution and use is allowed according to the terms of the BSD license.
+# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+
+find_package(PkgConfig)
+pkg_check_modules(PC_CANBERRA libcanberra)
+
+find_library(CANBERRA_LIBRARIES
+    NAMES canberra
+    HINTS ${PC_CANBERRA_LIBRARY_DIRS} ${PC_CANBERRA_LIBDIR}
+)
+
+find_path(CANBERRA_INCLUDE_DIRS
+    NAMES canberra.h
+    HINTS ${PC_CANBERRA_INCLUDE_DIRS} ${PC_CANBERRA_INCLUDEDIR}
+)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(Canberra REQUIRED_VARS CANBERRA_LIBRARIES CANBERRA_INCLUDE_DIRS)
+
+mark_as_advanced(CANBERRA_LIBRARIES CANBERRA_INCLUDE_DIRS)
diff -urN a/kmix/CMakeLists.txt b/kmix/CMakeLists.txt
--- a/kmix/CMakeLists.txt	2012-08-25 17:19:49.704288275 +0200
+++ b/kmix/CMakeLists.txt	2012-08-26 11:53:19.620054365 +0200
@@ -12,8 +12,8 @@
 
 # FIXME: Disable libcanberra support in the 4.8 branch because it is
 #        still under development in master.
-# pkg_check_modules(CANBERRA libcanberra)
-# macro_log_feature(CANBERRA_FOUND "libcanberra" "libcanberra audio library" "http://0pointer.de/lennart/projects/libcanberra/" FALSE "" "libcanberra is needed for kmix sound feedback")
+macro_optional_find_package(Canberra)
+macro_log_feature(CANBERRA_FOUND "libcanberra" "libcanberra audio library" "http://0pointer.de/lennart/projects/libcanberra/" FALSE "" "libcanberra is needed for kmix sound feedback")
 
 alsa_configure_file(${CMAKE_BINARY_DIR}/config-alsa.h)
 
